controller.js

```diff
@@ -78,12 +83,44 @@ exports.postComments = (req, res, next) => {
       message: "Bad Request: username and body are required"
     });
   };
-  fetchArticleById(article_id)
+  checkArticleExists(article_id)
   .then(() => {
-  return insertComment(article_id, username, body);
+    return insertComment(article_id, username, body);
   })
   .then((newComment) => {
     res.status(201).send({ comment: newComment });
   })
   .catch(next);
-};
\ No newline at end of file
+};
+
+// PATCH /api/articles/:article_id
+exports.patchArticle = (req, res, next) => {
```

Comment:
The controller should not be rejecting, that should all be handled in the model. At least three of these custom rejects would be handled by psql rejecting the query.